File: mobile/android/NexusApp/domain/src/main/kotlin/com/nexus/domain/repository/AuthRepository.kt
package com.nexus.domain.repository

import com.nexus.domain.entity.User

data class AuthTokens(
    val accessToken: String,
    val refreshToken: String,
    val expiresIn: Int
)

data class MFASetup(
    val secret: String,
    val qrCodeURL: String,
    val backupCodes: List<String>,
    val tempToken: String
)

interface AuthRepository {
    suspend fun login(email: String, password: String): Pair<User, AuthTokens>
    suspend fun register(email: String, password: String, firstName: String, lastName: String, tenantName: String): User
    suspend fun refreshToken(refreshToken: String): AuthTokens
    suspend fun logout(refreshToken: String)
    suspend fun setupMFA(): MFASetup
    suspend fun verifyMFA(code: String, tempToken: String): Pair<User, AuthTokens>
}

File: mobile/ios/NexusApp/Domain/RepositoryProtocols/AuthRepository.swift
import Foundation

protocol AuthRepository: Sendable {
    func login(email: String, password: String) async throws -> (User, AuthTokens)
    func register(email: String, password: String, firstName: String, lastName: String, tenantName: String) async throws -> User
    func refreshToken(_ refreshToken: String) async throws -> AuthTokens
    func logout(refreshToken: String) async throws
    func setupMFA() async throws -> MFASetup
    func verifyMFA(code: String, tempToken: String) async throws -> (User, AuthTokens)
}

struct AuthTokens: Sendable, Codable {
    let accessToken: String
    let refreshToken: String
    let expiresIn: Int
}

struct MFASetup: Sendable, Codable {
    let secret: String
    let qrCodeURL: String
    let backupCodes: [String]
    let tempToken: String
}